client/components/common/EnsureLoggedIn.js

```diff
@@ -0,0 +1,35 @@
+import React, {Component} from 'react';
```

Comment:
Instead of creating an actual component for this, can't we display a flash message on the LoginPage component to let the user try again. Or say he/she isn't authorized.

Something like this:
<img width="810" alt="screen shot 2017-11-21 at 12 28 09" src="https://user-images.githubusercontent.com/25608335/33070187-79e0ce1c-ceb7-11e7-9eef-3c2a57f348ac.png">

Reply:
Or what does this component do exactly, having trouble understanding what it's doing?
Since we can be checking if a user is unauthorized from the LoginPage component and not allow the user inside the application.

Reply:
The `EnsureLoggedIn` component acts as a protector route for all our routes that requires authentication. This makes it easy to do all the auth checking logic in one place rather than in individual routes. Also, this component sets the actual URL requested by an unauthenticated user so user can be directed back to that url after authentication. I think it is a good idea to add the `Unauthorized` flash message after an unauthorised user is redirected to the login page.